### PR TITLE
TE7-T1151: Add Search & Category

### PR DESCRIPTION
**Pull request by Kruna1Pate1**
- Add search with category
- User debounce
- Implement pagination

### CODE REVIEW

File: My Spotify/My Spotify/Modules/Search/Controllers/SearchHistoryVC.swift
Comment (DevarshBhalara) [inline My Spotify/My Spotify/Modules/Search/Controllers/SearchHistoryVC.swift:1]: Why you created two methods for one viewModel bind ?

Comment (DevarshBhalara) [inline My Spotify/My Spotify/Modules/Search/Controllers/SearchHistoryVC.swift:1]: Mark Tableview Delegate

File: My Spotify/My Spotify/Modules/Search/Controllers/SearchVC.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        
+        let category = searchViewModel.categories.value[indexPath.item]
+        
+        cell.configCell(category: category)
+        return cell
+    }
+    
+}
+
+extension SearchVC: UICollectionViewDelegate {
```
Comment (DevarshBhalara) [inline My Spotify/My Spotify/Modules/Search/Controllers/SearchVC.swift:71]: If you don't need collection view delegate then don't write

File: My Spotify/My Spotify/Modules/Search/Controllers/SearchHistoryVC.swift
Comment (Kruna1Pate1) [reply My Spotify/My Spotify/Modules/Search/Controllers/SearchHistoryVC.swift:1]: This is standard which is followed.

**Review by DevarshBhalara — APPROVED** [review]

**Pull request merged**